internal/utils/events.go

```diff
@@ -0,0 +1,117 @@
+package utils
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/Taraxa-project/taraxa-indexer/models"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// LogReward ..
+type LogReward struct {
+	Account   string
+	Validator string
+	Value     *big.Int
+	EventName string
+}
+
+type RewardsClaimedEvent struct {
+	Account   string
+	Validator string
+	Amount    *big.Int
+}
+
+type CommissionRewardsClaimedEvent struct {
+	Account   string
+	Validator string
+	Amount    *big.Int
+}
+
+const dposABI = `[{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"account","type":"address"},{"indexed":true,"internalType":"address","name":"validator","type":"address"},{"indexed":false,"internalType":"uint256","name":"amount","type":"uint256"}],"name":"RewardsClaimed","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"account","type":"address"},{"indexed":true,"internalType":"address","name":"validator","type":"address"},{"indexed":false,"internalType":"uint256","name":"amount","type":"uint256"}],"name":"CommissionRewardsClaimed","type":"event"},{"inputs":[{"internalType":"address","name":"validator","type":"address"}],"name":"claimCommissionRewards","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"address","name":"validator","type":"address"}],"name":"claimRewards","outputs":[],"stateMutability":"nonpayable","type":"function"}]`
+
+func DecodeEvent(log models.EventLog) (interface{}, error) {
```

Comment:
I think you should also check address of event "sender" as we shouldn't care about other contracts. It can mess calculations if someone will implement contract with event like this in some other contract

Reply:
I'd like to keep this as is now because in the next [issue](https://github.com/Taraxa-project/taraxa-indexer/issues/94) I'd like to implement a generalized approach for all events

Reply:
Ok, I see that you check it in the `DecodeRewardsTopics`